### PR TITLE
feat: Added parameter `maximum_number_of_learner` and `learner` to `AdaBoost`

### LINKED ISSUE #171: Set maximum number of learners of `AdaBoost`
### Is your feature request related to a problem?

It's not possible to set the maximum number of learners of `AdaBoost`.

### Desired solution

* Add a new parameter `maximum_number_of_learners: int` to the initializer of `safeds.ml.classification.AdaBoost` and `safeds.ml.regression.AdaBoost`
* Raise a value error if `maximum_number_of_learners` < 1
* Pass it along to `n_estimators` of the wrapped `scikit-learn` model in the `fit` method

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by alex-senger**
Closes #171 and #173.

### Summary of Changes

feat: Added parameter `maximum_number_of_learner` to `AdaBoost`
feat: Added parameter `learner` to `AdaBoost`
feat: Added private abstract Method `_get_scikit_learner` in `Classifier` and `Regressor`
refactor: Improved error messages in `classification` and `regression`
docs: Improved docstrings for raised `ValueErrors` in `classification` and `regression`

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.2) status: ✅ [SUCCESS](https://github.com/Safe-DS/Stdlib/actions/runs/4891471964)

| Descriptor  |                                 Linter                                 |Files|Fixed|Errors|Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.2/descriptors/python_black)          |    4|    2|     0|0.62s       |
|✅ PYTHON    |[mypy](https://megalinter.io/6.22.2/descriptors/python_mypy)            |    4|     |     0|1.7s        |
|✅ PYTHON    |[ruff](https://megalinter.io/6.22.2/descriptors/python_ruff)            |    4|    2|     0|0.04s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.2/descriptors/repository_git_diff)|yes  |     |no    |0.02s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4891471964)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/ml/classical/classification/_ada_boost.py [hunk 1]
```diff
@@ -13,27 +13,34 @@
...
-    """Ada Boost classification.
+    """
+    Ada Boost classification.
 
     Parameters
     ----------
     learning_rate : float
         Weight applied to each classifier at each boosting iteration. A higher learning rate increases the contribution
         of each classifier. Has to be greater than 0.
+    maximum_number_of_learners: int
```
Comment (lars-reimann) [inline src/safeds/ml/classical/classification/_ada_boost.py:24]: For the same reason as [here](https://github.com/Safe-DS/Stdlib/pull/268/files#r1185937745), the parameters should be swapped. Same in other places `learning_rate` and `maximum_number_of_learners` occur.

File: src/safeds/ml/classical/classification/_ada_boost.py [hunk 2]
```diff
@@ -13,27 +13,34 @@
...
+        The maximum number of learners at which boosting is terminated. In case of perfect fit, the learning procedure
+        is stopped early. Has to be greater than 0.
 
     Raises
     ------
     ValueError
-        If the learning rate is less than or equal to 0.
+        If the learning rate or maximum_number_of_learners are less than or equal to 0
```
Comment (lars-reimann) [inline src/safeds/ml/classical/classification/_ada_boost.py:31]: We should consistently write either the meaning of a parameter (learning rate) or its name (maximum_number_of_learners). I'd vote for the latter since it's more obvious. Might also put the names of the parameters in backticks.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: src/safeds/ml/classical/classification/_gradient_boosting.py [hunk 3]
```diff
@@ -29,15 +29,15 @@ class GradientBoosting(Classifier):
     Raises
     ------
     ValueError
-        If `learning_rate` is non-positive or the `number_of_trees` is less than or equal to 0.
+        If `learning_rate` is non-positive or `number_of_trees` is less than or equal to 0.
```
Comment (lars-reimann) [inline src/safeds/ml/classical/classification/_gradient_boosting.py:32]: Maybe swap these as well.

**Comment by lars-reimann** [comment]
The title of the PR should also indicate that `learner` was added.

File: src/safeds/ml/classical/regression/_gradient_boosting.py [hunk 4]
```diff
@@ -28,15 +30,15 @@ class GradientBoosting(Regressor):
     Raises
     ------
     ValueError
-        If `learning_rate` is non-positive or the `number_of_trees` is less than or equal to 0.
+        If `learning_rate` is non-positive or `number_of_trees` is less than or equal to 0.
```
Comment (alex-senger) [inline src/safeds/ml/classical/regression/_gradient_boosting.py]:
```suggestion
        If `number_of_trees` is less than or equal to 0 or `learning_rate` is non-positive.
```

**Review by lars-reimann — APPROVED** [review]
